snippets/graph/mst.cpp

```diff
@@ -0,0 +1,39 @@
+struct edge{
+	int u,v,w;
+	edge (int _, int __, int ___) { u = _, v = __, w = ___;}
+};
+
+// returns ptr to vector of edges in the mst, reasonably fast
+// graph[i] = vector of edges incident to vertex i
+// places total weight of the mst in &total
+vector<edge> *mst(vector<edge> *graph, size_t n, size_t m, long long &total) {
+    total = 0;
+    const auto compEdge = [](edge e, edge f) { return e.w > f.w; };
+    priority_queue<edge,vector<edge>,decltype(compEdge)> pq(compEdge);
+    vector<edge> *MST = new vector<edge>;
+    bool marked[n];
+    memset(marked,false,n);
+	  marked[0] = true; // mark vertex 0 as visited
```

Comment:
could replace this with a bitset hopefully -- don't know how allocation performance compares but it's smaller in memory and likely better for cache